.github/workflows/masterbot.yml

```diff
@@ -0,0 +1,87 @@
+name: Master Robot
+on:
+  push:
+    branches:
+      - master
+    tags-ignore: ['*']
+jobs:
+  build:
+    name: "Build"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/setup
+  lint-check:
+    name: "Check with linter"
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/setup
+      - run: npm run lint:check
+  generate-doc:
+    name: "Generate doc"
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/setup
+      - run: npm run doc:generate
+  unit-test:
+    name: "Run unit tests"
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: ./.github/actions/setup
+      - run: npx --no-install lerna run copy:staticresources --scope @coveo/quantic
+      - run: npm test
+  e2e-atomic-test:
+    name: "Run e2e tests on Atomic"
```

Comment:
I didn't add the matrix/parallel tests to the `master` branch either, since it may be complicated to define a `strategy` depending on whether or not we're on the `master` branch, and I don't want to have too many issues on the `master` branch to fix once this is merged. I can do it in a separate PR, if you think it's important.